api/db-cdr/generate-cdr/validate-prerequisites-exist.sh

```diff
@@ -8,62 +8,97 @@ export BQ_DATASET=$2        # CDR dataset
 export CDR_VERSION=$3       # CDR version
 export BUCKET=$4            # Bucket
 
-FILE_DIR="csv"
+TEMP_FILE_DIR="csv"
+CSV_HOME_DIR="cdr_csv_files"
 CRITERIA_MENU="cb_criteria_menu.csv"
 PREP_CRITERIA="prep_criteria.csv"
 PREP_CRITERIA_ANCESTOR="prep_criteria_ancestor.csv"
 PREP_CLINICAL_TERMS="prep_clinical_terms_nc.csv"
-All_FILES=("$CRITERIA_MENU" "$PREP_CRITERIA" "$PREP_CRITERIA_ANCESTOR" "$PREP_CLINICAL_TERMS")
+All_FILES=($CRITERIA_MENU $PREP_CRITERIA $PREP_CRITERIA_ANCESTOR $PREP_CLINICAL_TERMS)
 INCOMPATIBLE_DATASETS=("R2019Q4R3" "R2019Q4R4")
 
-if [[ "${INCOMPATIBLE_DATASETS[@]}" =~ "$BQ_DATASET" ]];
+if [[ ${INCOMPATIBLE_DATASETS[@]} =~ $BQ_DATASET ]];
   then
   echo "Can't run CDR build indices against "$BQ_DATASET"!"
   exit 1
 fi
 
-rm -rf "$FILE_DIR"
-mkdir "$FILE_DIR"
+# Purge all backup csv files except for the last 10 versions
+fileCount=$(gsutil ls gs://all-of-us-workbench-private-cloudsql/DummySR/cdr_csv_files/backup | wc -l)
```

Comment:
Should this be on the basis of date rather than just last 10 versions? eg purge all files from last month?

Reply:
Sometimes we go long periods without running this command. If we do it will delete all of backups. I want to always keep the last 10 in case we mess something up.

Reply:
Makes sense, my only concern was that there will always be just one copy/last run in backup.

Reply:
My bad, i thought its going to delete 10 versions always, however this code is going to keep 10 versions and delete rest. I am good with that